From cf43277db894443dcf49314e24f4b6ddb4132333 Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Sun, 27 Mar 2022 23:09:00 +0800
Subject: [PATCH 2/2] fix uboot disable dtc selfbuilt

---
 scripts/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/Makefile b/scripts/Makefile
index ce641b9..e27308a 100644
--- a/scripts/Makefile
+++ b/scripts/Makefile
@@ -19,4 +19,4 @@ build_docproc: $(obj)/docproc
 
 # Let clean descend into subdirs
 subdir-	+= basic kconfig
-#subdir-$(CONFIG_DTC)	+= dtc
+subdir-$(CONFIG_DTC)	+= dtc
-- 
2.17.1

