From 7238dd81bf9236ae145f5033fdfb76e5126563a2 Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Sun, 27 Mar 2022 23:26:42 +0800
Subject: [PATCH] fix No rule to make target sunxi_challenge

---
 Makefile | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index f036d42..59fd48e 100644
--- a/Makefile
+++ b/Makefile
@@ -1445,7 +1445,16 @@ include/config/uboot.release: include/config/auto.conf FORCE
 # version.h and scripts_basic is processed / created.
 
 # Listed in dependency order
-PHONY += prepare archprepare prepare0 prepare1 prepare2 prepare3
+PHONY += prepare archprepare prepare0 prepare1 prepare2 prepare3 cfg
+
+CLEAN_FILES += board/sunxi/sunxi_challenge.c
+board/sunxi/sunxi_challenge.c:
+	@echo "  prepare sunxi_challenge..."
+	@dd if=/dev/urandom of=sunxi_challenge bs=128 count=1 > /dev/null 2>&1
+	@xxd -c 8 -i sunxi_challenge > board/sunxi/sunxi_challenge.c
+	@sed -i '/^unsigned/i __attribute__((__used__))' board/sunxi/sunxi_challenge.c
+	@rm sunxi_challenge
+prepare: board/sunxi/sunxi_challenge.c
 
 # prepare3 is used to check if we are building in a separate output directory,
 # and if so do:
-- 
2.17.1

