From dd487de4eb69704e64c693b27952116f87489acd Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Tue, 15 Mar 2022 12:21:36 +0800
Subject: [PATCH] add yuzuki ruler dtb

---
 arch/arm/dts/Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 30edc320..b90412de 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -601,7 +601,7 @@ dtb-$(CONFIG_MACH_SUN9I) += \
 	sun9i-a80-cx-a99.dtb
 
 dtb-$(CONFIG_MACH_SUNIV) += \
-	suniv-generic.dtb
+	suniv-f1c200s-uboot.dtb
 
 dtb-$(CONFIG_VF610) += vf500-colibri.dtb \
 	vf610-colibri.dtb \
-- 
2.17.1

