From 9f56f62b8f983f09288748ce7d21eef86260ee4a Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Tue, 24 May 2022 10:32:23 +0800
Subject: [PATCH] disable-uboot-dts-loc

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 7c034d7..38073ae 100644
--- a/Makefile
+++ b/Makefile
@@ -980,7 +980,7 @@ endif
 
 BOARD_DTS_EXIST = $(shell if [ -f $(DTS_PATH)/$(BOARD_DTS_NAME).dts ]; then echo yes; else echo no; fi;)
 
-DEVICE_BOARD_DTS_EXIST = $(shell if [ -f $(LICHEE_BOARD_CONFIG_DIR)/uboot-board.dts ]; then echo yes; else echo no; fi;)
+DEVICE_BOARD_DTS_EXIST = $(shell if [ -f $(LICHEE_BOARD_CONFIG_DIR)/sun8i-mangopi-mq-dual-uboot.dts ]; then echo yes; else echo no; fi;)
 
 DTS_WARNNING_SKIP :=	-W no-unit_address_vs_reg \
 			-W no-unit_address_format \
@@ -988,7 +988,7 @@ DTS_WARNNING_SKIP :=	-W no-unit_address_vs_reg \
 			-W no-pwms_property
 ifeq (x$(DEVICE_BOARD_DTS_EXIST), xyes)
 # add depend on external dts, make sure dts in uboot up to date
-dts/dt.dtb: $(LICHEE_BOARD_CONFIG_DIR)/uboot-board.dts
+dts/dt.dtb: $(LICHEE_BOARD_CONFIG_DIR)/sun8i-mangopi-mq-dual-uboot.dts
 endif
 
 all:		$(ALL-y) cfg
-- 
2.17.1

