diff -urN linux-6.0-orig/drivers/clk/sunxi-ng/ccu-sun20i-d1.c linux-6.0/drivers/clk/sunxi-ng/ccu-sun20i-d1.c
--- linux-6.0-orig/drivers/clk/sunxi-ng/ccu-sun20i-d1.c	2022-10-02 23:09:07.000000000 +0200
+++ linux-6.0/drivers/clk/sunxi-ng/ccu-sun20i-d1.c	2022-10-05 11:10:29.993729952 +0200
@@ -232,7 +232,7 @@
 	{ .hw = &pll_periph0_800M_clk.common.hw },
 };
 static SUNXI_CCU_MUX_DATA(cpux_clk, "cpux", cpux_parents,
-			  0x500, 24, 3, CLK_SET_RATE_PARENT);
+			  0x500, 24, 3, CLK_SET_RATE_PARENT | CLK_IS_CRITICAL);
 
 static const struct clk_hw *cpux_hws[] = { &cpux_clk.common.hw };
 static SUNXI_CCU_M_HWS(cpux_axi_clk, "cpux-axi",
