From 9d85a3fe0331f9091eb9f3ef2517002aec284a43 Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Fri, 25 Mar 2022 16:41:41 +0800
Subject: [PATCH 15/93] fix ramfs-input

---
 usr/Makefile | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/usr/Makefile b/usr/Makefile
index e6f7cb2f8..9a3988940 100644
--- a/usr/Makefile
+++ b/usr/Makefile
@@ -30,6 +30,11 @@ hostprogs-y := gen_init_cpio
 initramfs   := $(CONFIG_SHELL) $(srctree)/$(src)/gen_initramfs_list.sh
 ramfs-input := $(if $(filter-out "",$(CONFIG_INITRAMFS_SOURCE)), \
 			$(shell echo $(CONFIG_INITRAMFS_SOURCE)),-d)
+
+ifeq ($(filter /% -d,$(ramfs-input)),)
+ramfs-input := $(patsubst %, $(srctree)/%, $(ramfs-input))
+endif
+
 ramfs-args  := \
         $(if $(CONFIG_INITRAMFS_ROOT_UID), -u $(CONFIG_INITRAMFS_ROOT_UID)) \
         $(if $(CONFIG_INITRAMFS_ROOT_GID), -g $(CONFIG_INITRAMFS_ROOT_GID))
-- 
2.17.1

