From f7af93a7ce1fc7619a4b9e38339305474a22c6f1 Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Fri, 25 Mar 2022 17:16:42 +0800
Subject: [PATCH 68/93] tristate to bool

---
 drivers/vfio/Kconfig | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/vfio/Kconfig b/drivers/vfio/Kconfig
index fd17db9b4..2976ad5f4 100644
--- a/drivers/vfio/Kconfig
+++ b/drivers/vfio/Kconfig
@@ -20,7 +20,7 @@ config VFIO_VIRQFD
 	default n
 
 menuconfig VFIO
-	tristate "VFIO Non-Privileged userspace driver framework"
+	bool "VFIO Non-Privileged userspace driver framework"
 	depends on IOMMU_API
 	select VFIO_IOMMU_TYPE1 if (X86 || S390 || ARM || ARM64)
 	help
-- 
2.17.1

