From 6d46940a908630c8a3a0f32018b46aaabe813f43 Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Fri, 25 Mar 2022 17:18:36 +0800
Subject: [PATCH 72/93] add driver for kconfig

---
 drivers/Kconfig | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/Kconfig b/drivers/Kconfig
index 8befa53f4..e7d1e8ad2 100644
--- a/drivers/Kconfig
+++ b/drivers/Kconfig
@@ -180,6 +180,8 @@ source "drivers/vme/Kconfig"
 
 source "drivers/pwm/Kconfig"
 
+source "drivers/dsp/Kconfig"
+
 source "drivers/irqchip/Kconfig"
 
 source "drivers/ipack/Kconfig"
@@ -200,6 +202,8 @@ source "drivers/thunderbolt/Kconfig"
 
 source "drivers/android/Kconfig"
 
+source "drivers/gpu/trace/Kconfig"
+
 source "drivers/nvdimm/Kconfig"
 
 source "drivers/dax/Kconfig"
@@ -214,6 +218,8 @@ source "drivers/fsi/Kconfig"
 
 source "drivers/tee/Kconfig"
 
+source "drivers/sunxi_drm_heap/Kconfig"
+
 source "drivers/mux/Kconfig"
 
 source "drivers/opp/Kconfig"
-- 
2.17.1

