From 8a4537d46121fc628482731cbbc9680ad8cbb97e Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Fri, 25 Mar 2022 17:22:38 +0800
Subject: [PATCH 76/93] add block driver

---
 drivers/block/Makefile | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/block/Makefile b/drivers/block/Makefile
index a53cc1e3a..827da8122 100644
--- a/drivers/block/Makefile
+++ b/drivers/block/Makefile
@@ -41,5 +41,8 @@ obj-$(CONFIG_BLK_DEV_NULL_BLK)	+= null_blk.o
 null_blk-objs	:= null_blk_main.o
 null_blk-$(CONFIG_BLK_DEV_ZONED) += null_blk_zoned.o
 
+
+
+
 skd-y		:= skd_main.o
 swim_mod-y	:= swim.o swim_asm.o
-- 
2.17.1

