From 4d5725d575722bda5782d3f8cdd6953a88d59e5c Mon Sep 17 00:00:00 2001
From: YuzukiTsuru <gloomyghost@gloomyghost.com>
Date: Sun, 6 Mar 2022 23:14:07 +0800
Subject: [PATCH 22/22] arch: dts rename to suniv-uboot.dtb

---
 arch/arm/dts/Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 4130af76..30edc320 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -601,8 +601,7 @@ dtb-$(CONFIG_MACH_SUN9I) += \
 	sun9i-a80-cx-a99.dtb
 
 dtb-$(CONFIG_MACH_SUNIV) += \
-	suniv-f1c100s-generic.dtb \
-	suniv-f1c200s-generic.dtb
+	suniv-uboot.dtb
 
 dtb-$(CONFIG_VF610) += vf500-colibri.dtb \
 	vf610-colibri.dtb \
-- 
2.17.1

